Dfe.Academies.External.Web/Controllers/BaseController.cs

```diff
@@ -8,44 +8,30 @@ public class BaseController : Controller
     {
         public const int SearchQueryMinLength = 3;
         private readonly ILogger<BaseController> _logger;
-        public readonly IReferenceDataRetrievalService _referenceDataRetrievalService;
-        public readonly IConversionApplicationRetrievalService _conversionApplicationRetrievalService;
+        public readonly IReferenceDataRetrievalService ReferenceDataRetrievalService;
+        public readonly IConversionApplicationRetrievalService ConversionApplicationRetrievalService;
 
         public BaseController(ILogger<BaseController> logger,
                                 IReferenceDataRetrievalService referenceDataRetrievalService,
                                 IConversionApplicationRetrievalService conversionApplicationRetrievalService)
         {
             _logger = logger;
-            _referenceDataRetrievalService = referenceDataRetrievalService;
-            _conversionApplicationRetrievalService = conversionApplicationRetrievalService; 
+            ReferenceDataRetrievalService = referenceDataRetrievalService;
+            ConversionApplicationRetrievalService = conversionApplicationRetrievalService; 
         }
 
-        protected IActionResult CatchErrorAndRedirect(Exception ex)
+        protected async Task<ApplicationCacheValuesViewModel?> LoadAndSetApplicationDetails(int applicationId)
         {
-            _logger.LogError("BaseController::CatchErrorAndRedirect::Exception - {Message}", ex.Message);
-            return RedirectToAction("Error", "academies");
-        }
-
-        protected async Task<ApplicationCacheValuesViewModel> LoadAndSetApplicationDetails(int applicationId)
-        {
-            ApplicationCacheValuesViewModel cachedValuesViewModel = null;
-            var applicationDetails = await _conversionApplicationRetrievalService.GetApplication(applicationId, Enums.ApplicationTypes.FormNewMat);
+            ApplicationCacheValuesViewModel? cachedValuesViewModel = null;
+            var applicationDetails = await ConversionApplicationRetrievalService.GetApplication(applicationId, Enums.ApplicationTypes.FormNewMat);
 
-            if (applicationDetails != null)
+            if (Object.ReferenceEquals(applicationDetails, null))
```

Comment:
Is this change to get rid of a warning?

Reply:
Yes, this removes a null reference exception warning